### PR TITLE
Identity function

### PR DESCRIPTION
**Pull request by fnandot**
Added identity function [https://en.wikipedia.org/wiki/Identity_function](https://en.wikipedia.org/wiki/Identity_function).

Identity is a function that always returns the same value that was used as its argument.

### CODE REVIEW

File: src/identity.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * @param mixed $value any type of value
+ *
+ * @return \Closure
+ */
+function identity($value)
+{
+    return function () use ($value) {
+        return $value;
+    };
+}
```
Comment (rgomezcasas) [inline src/identity.php]:
```php
function identity($value)
{
    return $value;
}
```

? :)

identity = `f(x) = x`
Here we have: `f(x) = f() -> x`

File: docs/functions/identity.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+use function Lambdish\Phunctional\map;
+use function Lambdish\Phunctional\identity;
+ 
+$numbers = [1,2,3,4,5];
+
+map(identity(0), $numbers);
+
+// => [0,0,0,0,0]
+```
```
Comment (rgomezcasas) [inline docs/functions/identity.md]:
```php
map('identity', $numbers)

// => [1, 2, 3, 4, 5]
```

**Comment by rgomezcasas** [comment]
Also, you can rename this to `constant` (https://www.youtube.com/watch?v=olR031-wcMs).

Buuuut! You can rename it to `constant` and then create another PR for the `identity` function :)

**Comment by fnandot** [comment]
So, constant is as I initially named it. :trollface:

File: docs/functions/identity.md (hunk 2 above)
Comment (fnandot) [reply docs/functions/identity.md]: That will be the identity function, I will create another PR with the real identity function.

File: src/identity.php (hunk 1 above)
Comment (fnandot) [reply src/identity.php]: constant = f() = c

**Pull request closed without merging**